File: benchmarks/python/pipeline.py
def double(x):
    return x * 2

def add_one(x):
    return x + 1

def square(x):
    return x * x

acc = 0
for n in range(10000000, 0, -1):
    acc += square(add_one(double(n)))
print(acc)

File: benchmarks/rust/pipeline.rs
fn double(x: i64) -> i64 { x * 2 }
fn add_one(x: i64) -> i64 { x + 1 }
fn square(x: i64) -> i64 { x.wrapping_mul(x) }

fn main() {
    let mut acc: i64 = 0;
    for n in (1..=10_000_000i64).rev() {
        acc = acc.wrapping_add(square(add_one(double(n))));
    }
    println!("{}", acc);
}
